zerver/tests/test_markdown.py

```diff
@@ -1156,6 +1158,58 @@ def test_fetch_tweet_data_settings_validation(self) -> None:
         with self.settings(TEST_SUITE=False, TWITTER_CONSUMER_KEY=None):
             self.assertIs(None, fetch_tweet_data("287977969287315459"))
 
+    def mocked_requests_post(*args: Any, **kwargs: Any) -> MockPythonResponse:
+        """This method will be used by the mock to replace requests.post"""
+
+        json_data = kwargs.get("json", {})
+        if not all(json_data.values()):
+            return MockPythonResponse("", 400)
+
+        return MockPythonResponse(args[1], 200)
```

Comment:
We should use the `responses` module for mocking HTTP responses, rather than a custom system.